# *Merge* entre as tabelas

In [2]:
from IPython.display import display
import pandas as pd

df1 = pd.read_excel("ocorrencia1_aeronauticos.xlsx")
df2 = pd.read_excel("Aeronave.xlsx")
df3 = pd.read_excel("ocorrencia_tipo.xlsx")
df4 = pd.read_excel("Fator_contribuinte.xlsx")

tb1 = pd.merge(df1, df2, how = 'inner', on = 'codigo_ocorrencia')
tb1 = pd.merge(tb1, df3, how = 'inner', on = 'codigo_ocorrencia')
tb1 = pd.merge(tb1, df4, how = 'inner', on = 'codigo_ocorrencia')
tb1.to_excel("tabelao.xlsx")

In [3]:
display(tb1)

codigo_ocorrencia ocorrencia_classificacao   ocorrencia_cidade  \
0                 43628                 ACIDENTE               CAIRU   
1                 43628                 ACIDENTE               CAIRU   
2                 43628                 ACIDENTE               CAIRU   
3                 43628                 ACIDENTE               CAIRU   
4                 43657                 ACIDENTE         DOM PEDRITO   
...                 ...                      ...                 ...   
5171              80365                 ACIDENTE  SÃO JOSÉ DE MIPIBU   
5172              80383                 ACIDENTE   LAGOA DA CONFUSÃO   
5173              80383                 ACIDENTE   LAGOA DA CONFUSÃO   
5174              80383                 ACIDENTE   LAGOA DA CONFUSÃO   
5175              80383                 ACIDENTE   LAGOA DA CONFUSÃO   

     ocorrencia_uf ocorrencia_dia ocorrencia_hora  \
0               BA     2011-01-05        19:40:00   
1               BA     2011-01-05        19:40:00   
2               BA     2011-01-05        19:40:00   
3               BA     2011-01-05        19:40:00   
4               RS     2011-01-11        19:30:00   
...            ...            ...             ...   
5171            RN     2021-11-07        19:08:00   
5172            TO     2021-11-11        12:40:00   
5173            TO     2021-11-11        12:40:00   
5174            TO     2021-11-11        12:40:00   
5175            TO     2021-11-11        12:40:00   

     investigacao_aeronave_liberada investigacao_status  \
0                               ***          FINALIZADA   
1                               ***          FINALIZADA   
2                               ***          FINALIZADA   
3                               ***          FINALIZADA   
4                               ***          FINALIZADA   
...                             ...                 ...   
5171                            SIM          FINALIZADA   
5172                            SIM          FINALIZADA   
5173                            SIM          FINALIZADA   
5174                            SIM          FINALIZADA   
5175                            SIM          FINALIZADA   

     divulgacao_relatorio_numero divulgacao_relatorio_publicado  ...  \
0              A-584/CENIPA/2016                            SIM  ...   
1              A-584/CENIPA/2016                            SIM  ...   
2              A-584/CENIPA/2016                            SIM  ...   
3              A-584/CENIPA/2016                            SIM  ...   
4              A-072/CENIPA/2012                            SIM  ...   
...                          ...                            ...  ...   
5171           A-124/CENIPA/2021                            SIM  ...   
5172           A-126/CENIPA/2021                            SIM  ...   
5173           A-126/CENIPA/2021                            SIM  ...   
5174           A-126/CENIPA/2021                            SIM  ...   
5175           A-126/CENIPA/2021                            SIM  ...   

     aeronave_fase_operacao  aeronave_tipo_operacao  aeronave_nivel_dano  \
0                     POUSO              TÁXI AÉREO          SUBSTANCIAL   
1                     POUSO              TÁXI AÉREO          SUBSTANCIAL   
2                     POUSO              TÁXI AÉREO          SUBSTANCIAL   
3                     POUSO              TÁXI AÉREO          SUBSTANCIAL   
4                 DECOLAGEM                AGRÍCOLA          SUBSTANCIAL   
...                     ...                     ...                  ...   
5171     ARREMETIDA NO SOLO                 PRIVADA            DESTRUÍDA   
5172                MANOBRA                AGRÍCOLA          SUBSTANCIAL   
5173                MANOBRA                AGRÍCOLA          SUBSTANCIAL   
5174                MANOBRA                AGRÍCOLA          SUBSTANCIAL   
5175                MANOBRA                AGRÍCOLA          SUBSTANCIAL   

     aeronave_fat

In [4]:
display(df4)

codigo_ocorrencia               fator_nome             fator_aspecto  \
0                 43628                   FADIGA            ASPECTO MÉDICO   
1                 43628                  MEMÓRIA       ASPECTO PSICOLÓGICO   
2                 43628  ORGANIZAÇÃO DO TRABALHO       ASPECTO PSICOLÓGICO   
3                 43628     SUPERVISÃO GERENCIAL  DESEMPENHO DO SER HUMANO   
4                 43657    APLICAÇÃO DE COMANDOS  DESEMPENHO DO SER HUMANO   
...                 ...                      ...                       ...   
4771              80365       PROCESSO DECISÓRIO       ASPECTO PSICOLÓGICO   
4772              80383                  ATITUDE       ASPECTO PSICOLÓGICO   
4773              80383  JULGAMENTO DE PILOTAGEM  DESEMPENHO DO SER HUMANO   
4774              80383      PLANEJAMENTO DE VOO  DESEMPENHO DO SER HUMANO   
4775              80383     SUPERVISÃO GERENCIAL  DESEMPENHO DO SER HUMANO   

       fator_condicionante         fator_area  
0                      ***       FATOR HUMANO  
1               INDIVIDUAL       FATOR HUMANO  
2           ORGANIZACIONAL       FATOR HUMANO  
3     OPERAÇÃO DA AERONAVE  FATOR OPERACIONAL  
4     OPERAÇÃO DA AERONAVE  FATOR OPERACIONAL  
...                    ...                ...  
4771            INDIVIDUAL       FATOR HUMANO  
4772            INDIVIDUAL       FATOR HUMANO  
4773  OPERAÇÃO DA AERONAVE  FATOR OPERACIONAL  
4774  OPERAÇÃO DA AERONAVE  FATOR OPERACIONAL  
4775  OPERAÇÃO DA AERONAVE  FATOR OPERACIONAL  

[4776 rows x 5 columns]

## Análise geral

In [5]:
tb1_no_code = tb1.drop('codigo_ocorrencia', axis=1)
tb1_no_code.describe()

total_recomendacoes  total_aeronaves_envolvidas   aeronave_pmd  \
count          5176.000000                 5176.000000    5176.000000   
mean              1.954019                    1.027821    6943.551971   
std               2.496338                    0.164475   25108.306978   
min               0.000000                    1.000000       0.000000   
25%               0.000000                    1.000000    1157.000000   
50%               1.000000                    1.000000    1800.000000   
75%               3.000000                    1.000000    2717.000000   
max              23.000000                    2.000000  346544.000000   

       aeronave_assentos  aeronave_ano_fabricacao  aeronave_fatalidades_total  
count        5102.000000              5107.000000                 5176.000000  
mean           12.265778              1941.412963                    0.756376  
std            36.645431               307.957331                    1.779855  
min             0.000000                 0.000000                    0.000000  
25%             2.000000              1977.000000                    0.000000  
50%             4.000000              1991.000000                    0.000000  
75%             6.000000              2005.000000                    1.000000  
max           384.000000              2018.000000                   16.000000

In [6]:
tb1_no_code.corr()

<ipython-input-6-5e65de2b4dcd>:1: FutureWarning: The default value of numeric_only in DataFrame.corr is deprecated. In a future version, it will default to False. Select only valid columns or specify the value of numeric_only to silence this warning.
  tb1_no_code.corr()


total_recomendacoes  total_aeronaves_envolvidas  \
total_recomendacoes                    1.000000                    0.139601   
total_aeronaves_envolvidas             0.139601                    1.000000   
aeronave_pmd                           0.186860                    0.161067   
aeronave_assentos                      0.197094                    0.209676   
aeronave_ano_fabricacao               -0.046420                   -0.101897   
aeronave_fatalidades_total             0.407296                   -0.058694   

                            aeronave_pmd  aeronave_assentos  \
total_recomendacoes             0.186860           0.197094   
total_aeronaves_envolvidas      0.161067           0.209676   
aeronave_pmd                    1.000000           0.706623   
aeronave_assentos               0.706623           1.000000   
aeronave_ano_fabricacao        -0.156636           0.056301   
aeronave_fatalidades_total     -0.068328          -0.059745   

                            aeronave_ano_fabricacao  \
total_recomendacoes                       -0.046420   
total_aeronaves_envolvidas                -0.101897   
aeronave_pmd                              -0.156636   
aeronave_assentos                          0.056301   
aeronave_ano_fabricacao                    1.000000   
aeronave_fatalidades_total                 0.038312   

                            aeronave_fatalidades_total  
total_recomendacoes                           0.407296  
total_aeronaves_envolvidas                   -0.058694  
aeronave_pmd                                 -0.068328  
aeronave_assentos                            -0.059745  
aeronave_ano_fabricacao                       0.038312  
aeronave_fatalidades_total                    1.000000

### Análise por localização e período

In [7]:
import matplotlib.pyplot as plt
import plotly.express as px

In [8]:
df_local = pd.DataFrame(tb1, columns=['ocorrencia_cidade', 'ocorrencia_uf'])
# Realiza a contagem com base nas colunas de cidade e estado
count_cities = df_local.groupby(['ocorrencia_cidade', 'ocorrencia_uf']).value_counts().reset_index(name='Ocorrências') # redefinindo o índice
count_cities = count_cities.rename(columns={'ocorrencia_cidade': 'Cidade','ocorrencia_uf': 'Estado'}).sort_values(by='Ocorrências', ascending=False)
count_uf = df_local['ocorrencia_uf'].value_counts().reset_index() # Retorna uma série contendo contagens de valores únicos.
count_uf = count_uf.rename(columns={'index': 'Estado', 'ocorrencia_uf': 'Ocorrências'})

# Exibe o resultado com aliases para cada coluna
print("Ocorrências por cidade:\n")
print(count_cities.head(10).to_string(index=False))
print("\n\n\nOcorrências por Estado:\n")
print(count_uf.head(10).to_string(index=False))

Ocorrências por cidade:

        Cidade Estado  Ocorrências
RIO DE JANEIRO     RJ          161
        MANAUS     AM          120
      BRASÍLIA     DF          104
     SÃO PAULO     SP           90
       GOIÂNIA     GO           87
        RECIFE     PE           78
       ATIBAIA     SP           59
         BELÉM     PA           58
       JUNDIAÍ     SP           55
      ITAITUBA     PA           54



Ocorrências por Estado:

Estado  Ocorrências
    SP          995
    RS          500
    PR          379
    MT          364
    PA          353
    GO          341
    AM          301
    MG          296
    RJ          266
    BA          259


Neste código, utilizamos o método `value_counts()` para contar as ocorrências por cidade e por estado. O método `reset_index()` é utilizado para redefenir o índice e assim ordenar adequadamente conforme o número de ocorrências.

Ao imprimir o resultado, usamos o método `to_string()` para exibir o DataFrame com os cabeçalhos especificados.



In [9]:
# Plotar o gráfico de barras
fig = px.bar(count_cities.head(10), x='Cidade', y='Ocorrências', labels={'index': 'Cidade', 'ocorrencia_cidade': 'Número de Ocorrências'},
             title='Número de Ocorrências por Cidade', width=1000, height=700)
fig.update_xaxes(tickangle=45)

fig.show()

### Análise por faixa de hora

In [10]:
display(tb1.head())

codigo_ocorrencia ocorrencia_classificacao ocorrencia_cidade ocorrencia_uf  \
0              43628                 ACIDENTE             CAIRU            BA   
1              43628                 ACIDENTE             CAIRU            BA   
2              43628                 ACIDENTE             CAIRU            BA   
3              43628                 ACIDENTE             CAIRU            BA   
4              43657                 ACIDENTE       DOM PEDRITO            RS   

  ocorrencia_dia ocorrencia_hora investigacao_aeronave_liberada  \
0     2011-01-05        19:40:00                            ***   
1     2011-01-05        19:40:00                            ***   
2     2011-01-05        19:40:00                            ***   
3     2011-01-05        19:40:00                            ***   
4     2011-01-11        19:30:00                            ***   

  investigacao_status divulgacao_relatorio_numero  \
0          FINALIZADA           A-584/CENIPA/2016   
1          FINALIZADA           A-584/CENIPA/2016   
2          FINALIZADA           A-584/CENIPA/2016   
3          FINALIZADA           A-584/CENIPA/2016   
4          FINALIZADA           A-072/CENIPA/2012   

  divulgacao_relatorio_publicado  ... aeronave_fase_operacao  \
0                            SIM  ...                  POUSO   
1                            SIM  ...                  POUSO   
2                            SIM  ...                  POUSO   
3                            SIM  ...                  POUSO   
4                            SIM  ...              DECOLAGEM   

   aeronave_tipo_operacao  aeronave_nivel_dano aeronave_fatalidades_total  \
0              TÁXI AÉREO          SUBSTANCIAL                        0.0   
1              TÁXI AÉREO          SUBSTANCIAL                        0.0   
2              TÁXI AÉREO          SUBSTANCIAL                        0.0   
3              TÁXI AÉREO          SUBSTANCIAL                        0.0   
4                AGRÍCOLA          SUBSTANCIAL                        0.0   

            ocorrencia_tipo taxonomia_tipo_icao               fator_nome  \
0            POUSO SEM TREM                 ARC                   FADIGA   
1            POUSO SEM TREM                 ARC                  MEMÓRIA   
2            POUSO SEM TREM                 ARC  ORGANIZAÇÃO DO TRABALHO   
3            POUSO SEM TREM                 ARC     SUPERVISÃO GERENCIAL   
4  PERDA DE CONTROLE EM VOO               LOC-I    APLICAÇÃO DE COMANDOS   

              fator_aspecto   fator_condicionante         fator_area  
0            ASPECTO MÉDICO                   ***       FATOR HUMANO  
1       ASPECTO PSICOLÓGICO            INDIVIDUAL       FATOR HUMANO  
2       ASPECTO PSICOLÓGICO        ORGANIZACIONAL       FATOR HUMANO  
3  DESEMPENHO DO SER HUMANO  OPERAÇÃO DA AERONAVE  FATOR OPERACIONAL  
4  DESEMPENHO DO SER HUMANO  OPERAÇÃO DA AERONAVE  FATOR OPERACIONAL  

[5 rows x 35 columns]

In [11]:
df_data = pd.DataFrame(tb1, columns=['ocorrencia_dia', 'ocorrencia_hora'])

print(df_data)

     ocorrencia_dia ocorrencia_hora
0        2011-01-05        19:40:00
1        2011-01-05        19:40:00
2        2011-01-05        19:40:00
3        2011-01-05        19:40:00
4        2011-01-11        19:30:00
...             ...             ...
5171     2021-11-07        19:08:00
5172     2021-11-11        12:40:00
5173     2021-11-11        12:40:00
5174     2021-11-11        12:40:00
5175     2021-11-11        12:40:00

[5176 rows x 2 columns]


In [12]:
count_time = df_data.groupby(['ocorrencia_dia', 'ocorrencia_hora']).value_counts().reset_index(name='Ocorrências')
count_time = count_time.rename(columns={'ocorrencia_dia': 'Dia','ocorrencia_hora': 'Horário'}).sort_values(by='Ocorrências', ascending=False)

print("Ocorrências em cada horário:\n")
print(count_time.head(10).to_string(index=False))

Ocorrências em cada horário:

       Dia  Horário  Ocorrências
2017-11-30 08:37:00           26
2018-04-09 00:32:00           26
2012-12-05 12:30:00           20
2019-01-02 12:20:00           20
2018-11-03 22:08:00           18
2011-06-17 21:49:00           18
2011-07-13 09:54:00           18
2018-11-30 12:20:00           18
2015-06-26 19:38:00           17
2016-01-21 12:58:00           16


In [42]:
from datetime import time
# Converter a coluna 'ocorrencia_hora' para representação numérica
df_data['Hora_Numérica'] = df_data['ocorrencia_hora'].apply(lambda t: t.hour + t.minute / 60 + t.second / 3600)

# Criar faixas de horário
faixas_horario = []
for i in range(24):
    faixas_horario.append(f"{i:02d}:00:01 - {i+1:02d}:00:00")

# Agrupar as ocorrências nas faixas de horário e calcular a contagem
df_data['Faixa_Horario'] = pd.cut(df_data['Hora_Numérica'], bins=range(24), labels=faixas_horario[:-1]) # labels faz a formatação conforme o laço de repetição acima
count_faixas = df_data.groupby('Faixa_Horario')['ocorrencia_hora'].count().reset_index(name='Ocorrências').sort_values(by='Ocorrências', ascending=False)

# Imprimir o resultado
print("Ocorrências em cada faixa de horário:\n")
print(count_faixas.head(10).to_string(index=False))

Ocorrências em cada faixa de horário:

      Faixa_Horario  Ocorrências
19:00:01 - 20:00:00          527
12:00:01 - 13:00:00          500
11:00:01 - 12:00:00          473
13:00:01 - 14:00:00          433
18:00:01 - 19:00:00          406
15:00:01 - 16:00:00          383
17:00:01 - 18:00:00          375
14:00:01 - 15:00:00          352
20:00:01 - 21:00:00          335
21:00:01 - 22:00:00          312


In [41]:
df_data_ordered = df_data.sort_values(by='Faixa_Horario', ascending=True)
px.histogram(df_data_ordered['Faixa_Horario'])

In [15]:

fig = px.bar(count_faixas.head(10), x='Faixa_Horario', y='Ocorrências', labels={'index': 'Horário', 'ocorrencia_hora': 'Número de Ocorrências'},
             title='Horários com mais ocorrências', width=1200, height=600)
fig.update_xaxes(tickangle=45)

fig.show()

## Análise por Ano

In [16]:
display(df_data)

ocorrencia_dia ocorrencia_hora  Hora_Numérica        Faixa_Horario
0        2011-01-05        19:40:00      19.666667  19:00:01 - 20:00:00
1        2011-01-05        19:40:00      19.666667  19:00:01 - 20:00:00
2        2011-01-05        19:40:00      19.666667  19:00:01 - 20:00:00
3        2011-01-05        19:40:00      19.666667  19:00:01 - 20:00:00
4        2011-01-11        19:30:00      19.500000  19:00:01 - 20:00:00
...             ...             ...            ...                  ...
5171     2021-11-07        19:08:00      19.133333  19:00:01 - 20:00:00
5172     2021-11-11        12:40:00      12.666667  12:00:01 - 13:00:00
5173     2021-11-11        12:40:00      12.666667  12:00:01 - 13:00:00
5174     2021-11-11        12:40:00      12.666667  12:00:01 - 13:00:00
5175     2021-11-11        12:40:00      12.666667  12:00:01 - 13:00:00

[5176 rows x 4 columns]

In [17]:
# Extrair o ano da coluna 'ocorrencia_dia'
df_data['Ano'] = pd.to_datetime(df_data['ocorrencia_dia']).dt.year

# Imprimir o resultado
print("Ano das ocorrências:\n")
print(df_data['Ano'])

Ano das ocorrências:

0       2011
1       2011
2       2011
3       2011
4       2011
        ... 
5171    2021
5172    2021
5173    2021
5174    2021
5175    2021
Name: Ano, Length: 5176, dtype: int64


In [18]:
count_year = df_data['Ano'].value_counts().reset_index(name='Ocorrências')
count_year = count_year.rename(columns={'index': 'Ano'})

print("Ocorrências em cada ano:\n")
print(count_year.head(10).to_string(index=False))

Ocorrências em cada ano:

 Ano  Ocorrências
2011          727
2012          640
2013          579
2014          570
2018          560
2015          492
2019          465
2017          436
2016          388
2020          280


In [19]:
fig = px.histogram(df_data['Ano'], width=1200)
fig.update_layout(bargap=0.15)
fig.show()

## Análise pelo destino

In [20]:
display(tb1)

codigo_ocorrencia ocorrencia_classificacao   ocorrencia_cidade  \
0                 43628                 ACIDENTE               CAIRU   
1                 43628                 ACIDENTE               CAIRU   
2                 43628                 ACIDENTE               CAIRU   
3                 43628                 ACIDENTE               CAIRU   
4                 43657                 ACIDENTE         DOM PEDRITO   
...                 ...                      ...                 ...   
5171              80365                 ACIDENTE  SÃO JOSÉ DE MIPIBU   
5172              80383                 ACIDENTE   LAGOA DA CONFUSÃO   
5173              80383                 ACIDENTE   LAGOA DA CONFUSÃO   
5174              80383                 ACIDENTE   LAGOA DA CONFUSÃO   
5175              80383                 ACIDENTE   LAGOA DA CONFUSÃO   

     ocorrencia_uf ocorrencia_dia ocorrencia_hora  \
0               BA     2011-01-05        19:40:00   
1               BA     2011-01-05        19:40:00   
2               BA     2011-01-05        19:40:00   
3               BA     2011-01-05        19:40:00   
4               RS     2011-01-11        19:30:00   
...            ...            ...             ...   
5171            RN     2021-11-07        19:08:00   
5172            TO     2021-11-11        12:40:00   
5173            TO     2021-11-11        12:40:00   
5174            TO     2021-11-11        12:40:00   
5175            TO     2021-11-11        12:40:00   

     investigacao_aeronave_liberada investigacao_status  \
0                               ***          FINALIZADA   
1                               ***          FINALIZADA   
2                               ***          FINALIZADA   
3                               ***          FINALIZADA   
4                               ***          FINALIZADA   
...                             ...                 ...   
5171                            SIM          FINALIZADA   
5172                            SIM          FINALIZADA   
5173                            SIM          FINALIZADA   
5174                            SIM          FINALIZADA   
5175                            SIM          FINALIZADA   

     divulgacao_relatorio_numero divulgacao_relatorio_publicado  ...  \
0              A-584/CENIPA/2016                            SIM  ...   
1              A-584/CENIPA/2016                            SIM  ...   
2              A-584/CENIPA/2016                            SIM  ...   
3              A-584/CENIPA/2016                            SIM  ...   
4              A-072/CENIPA/2012                            SIM  ...   
...                          ...                            ...  ...   
5171           A-124/CENIPA/2021                            SIM  ...   
5172           A-126/CENIPA/2021                            SIM  ...   
5173           A-126/CENIPA/2021                            SIM  ...   
5174           A-126/CENIPA/2021                            SIM  ...   
5175           A-126/CENIPA/2021                            SIM  ...   

     aeronave_fase_operacao  aeronave_tipo_operacao  aeronave_nivel_dano  \
0                     POUSO              TÁXI AÉREO          SUBSTANCIAL   
1                     POUSO              TÁXI AÉREO          SUBSTANCIAL   
2                     POUSO              TÁXI AÉREO          SUBSTANCIAL   
3                     POUSO              TÁXI AÉREO          SUBSTANCIAL   
4                 DECOLAGEM                AGRÍCOLA          SUBSTANCIAL   
...                     ...                     ...                  ...   
5171     ARREMETIDA NO SOLO                 PRIVADA            DESTRUÍDA   
5172                MANOBRA                AGRÍCOLA          SUBSTANCIAL   
5173                MANOBRA                AGRÍCOLA          SUBSTANCIAL   
5174                MANOBRA                AGRÍCOLA          SUBSTANCIAL   
5175                MANOBRA                AGRÍCOLA          SUBSTANCIAL   

     aeronave_fat

In [21]:
df_dest = pd.DataFrame(tb1, columns=['aeronave_voo_destino'])

print(df_dest)

     aeronave_voo_destino
0          FAZENDA PONTAL
1          FAZENDA PONTAL
2          FAZENDA PONTAL
3          FAZENDA PONTAL
4       FORA DE AERODROMO
...                   ...
5171     NÃO IDENTIFICADO
5172    FORA DE AERODROMO
5173    FORA DE AERODROMO
5174    FORA DE AERODROMO
5175    FORA DE AERODROMO

[5176 rows x 1 columns]


In [22]:
count_dest = df_dest['aeronave_voo_destino'].value_counts().reset_index() # Retorna uma série contendo contagens de valores únicos.
count_dest = count_dest.rename(columns={'index': 'Destino', 'aeronave_voo_destino': 'Ocorrências'})

# Exibe o resultado com aliases para cada coluna
print("\n\n\nOcorrências por Destino:\n")
print(count_dest.head(10).to_string(index=False))




Ocorrências por Destino:

                                 Destino  Ocorrências
                       FORA DE AERODROMO         1750
                        NÃO IDENTIFICADO           65
                          CAMPO DE MARTE           57
         PRESIDENTE JUSCELINO KUBITSCHEK           47
                            RONDONÓPOLIS           45
AERODRÓMO ESTADUAL DE CAMPINAS / AMARAIS           44
                               BACACHERI           42
                ESTADUAL ARTHUR SIQUEIRA           42
            GUARARAPES - GILBERTO FREYRE           40
           AERÓDROMO NACIONAL DE AVIAÇÃO           40


In [23]:
# Plotar o gráfico de barras
fig = px.bar(count_dest.head(10), x='Destino', y='Ocorrências', labels={'index': 'Destino', 'aeronave_voo_destino': 'Número de Ocorrências'},
             title='Número de Ocorrências por Cidade', width=1000, height=700)
fig.update_xaxes(tickangle=45)

fig.show()

# Análise por Modelo

In [24]:
import matplotlib.pyplot as plt
import plotly.express as px

Modelos com maior índice de ocorrências relacionadas

In [25]:
#é preciso definir o nome do dataframe, lembrando que é como se tivessemos criado uma nova tabela
df_mod = pd.DataFrame(tb1, columns=['aeronave_fatalidade_total', 'aeronave_modelo']) #nesse dataframe colocamos dentro

count_fatal = df_mod.groupby(['aeronave_fatalidade_total']).value_counts().reset_index(name='Ocorrências') # 'aeronave_modelo' redefinindo o índice

count_model = df_mod['aeronave_modelo'].value_counts().reset_index() # Retorna uma série contendo contagens de valores únicos.
count_model = count_model.rename(columns={'index': 'Modelos', 'aeronave_modelo': 'Ocorrências'})

# Exibe o resultado com aliases para cada coluna
print("\n\n A seguir podemos ver os 20 modelos de aeronave mais propensos a ter acidentes:\n")
print(count_model.head(20).to_string(index=False))



 A seguir podemos ver os 20 modelos de aeronave mais propensos a ter acidentes:

   Modelos  Ocorrências
    AB-115          222
  EMB-201A          216
   EMB-202          201
  EMB-202A          171
 AS 350 B2          149
     A188B          104
   AT-502B           96
    R44 II           95
        58           90
  EMB-810D           82
  EMB-810C           79
 PA-25-235           76
       152           74
  R22 BETA           72
  EMB-720D           71
      208B           70
      206B           67
PA-34-220T           66
   737-8EH           64
      56-C           53


In [26]:
plt.rcParams.update({'font.size': 20})

In [27]:
fig = px.bar(count_model.head(20), x='Modelos', y='Ocorrências', labels={'index': 'Modelos', 'aeronave_modelos': 'Ocorrências'},
             title='Número de Ocorrências por modelo', width=800, height=750)
fig.show()

Modelos com menor índice de ocorrências relacionadas, menor que 2

# Análise por Classificação de Veículo

Análise por tipo de avião- Avião, Helicoptero, ultra-leve e etc.

In [28]:
#é preciso definir o nome do dataframe, lembrando que é como se tivessemos criado uma nova tabela
df_tipoaer = pd.DataFrame(tb1, columns=['aeronave_tipo_veiculo']) #nesse dataframe colocamos dentro
#dele o conteudo das colunas que estão no parametro
# Usando o metodo value_counts podemos realizar a contagem com base nas colunas
count_tipo_real = df_tipoaer.groupby(['aeronave_tipo_veiculo']).value_counts().reset_index(name='Ocorrências') # redefinindo o índice
# count_tipo = count_tipo.rename(columns={'aeronave_fatalidade_total': 'Fatalidades','aeronave_modelo': 'Modelos'}).sort_values(by='Ocorrências', ascending=False)
count_tipo = df_tipoaer['aeronave_tipo_veiculo'].value_counts().reset_index() # Retorna uma série contendo contagens de valores únicos.
count_tipo = count_tipo.rename(columns={'index': 'Classificação Modelo', 'aeronave_tipo_veiculo': 'Ocorrências'})


print("\n\n A seguir podemos o total de ocorrências pela classificação do tipo de aeronave:\n")
print(count_tipo.head(10).to_string(index=False))



 A seguir podemos o total de ocorrências pela classificação do tipo de aeronave:

Classificação Modelo  Ocorrências
               AVIÃO         4101
         HELICÓPTERO          842
           ULTRALEVE          119
                 ***           58
            PLANADOR           35
             ANFÍBIO           21


In [29]:
fig = px.bar(count_tipo.head(20), x='Classificação Modelo', y='Ocorrências', labels={'index': 'Classificação Modelo', 'aeronave_modelos': 'Ocorrências'},
             title='Número de Ocorrências por classificação de modelo', width=800, height=750)
fig.show()



# Análise por tipo de falha

In [30]:
df_fator = pd.DataFrame(df4, columns=['fator_area', 'fator_nome'])
# Realiza a contagem com base nas colunas do tipo do fator e do fator em si
count_tipo_fator = df_fator['fator_area'].value_counts().reset_index()
count_tipo_fator = count_tipo_fator.rename(columns={'index': 'Tipo', 'fator_area': 'Ocorrências'})

count_fator = df_fator.groupby(['fator_area']).value_counts().reset_index(name='Ocorrências')
count_fator = count_fator.rename(columns={'fator_area': 'Tipo','fator_nome': 'Fator'}).sort_values(by=['Ocorrências'], ascending=False)



print("Ocorrências por tipo de fator:\n")
print(count_tipo_fator.head(20).to_string(index=False))

print("\n\n\n")

print("Ocorrências por fator específico:\n")
print(count_fator.head(10).to_string(index=False))




Ocorrências por tipo de fator:

             Tipo  Ocorrências
FATOR OPERACIONAL         2995
     FATOR HUMANO         1632
            OUTRO          116
   FATOR MATERIAL           33




Ocorrências por fator específico:

             Tipo                       Fator  Ocorrências
FATOR OPERACIONAL     JULGAMENTO DE PILOTAGEM          633
FATOR OPERACIONAL       APLICAÇÃO DE COMANDOS          441
FATOR OPERACIONAL         PLANEJAMENTO DE VOO          349
FATOR OPERACIONAL        SUPERVISÃO GERENCIAL          339
FATOR OPERACIONAL      MANUTENÇÃO DA AERONAVE          274
     FATOR HUMANO          PROCESSO DECISÓRIO          265
     FATOR HUMANO                     ATITUDE          242
FATOR OPERACIONAL POUCA EXPERIÊNCIA DO PILOTO          200
     FATOR HUMANO                   PERCEPÇÃO          177
FATOR OPERACIONAL         INDISCIPLINA DE VOO          136


Realizando a análise nessa parte dos dados, foi possível verificar que a maior parte dos acidentes são causados por fatores humanos. Tanto o tipo "FATOR HUMANO" quanto "FATOR OPERACIONAL" estão diretamente relacionados com a capacidade humana, pois dentro do "FATOR HUMANO" estão aspectos como cansaço, uso de medicamentos e no "FATOR OPERACIONAL" são encontrados fatores como má utilização de instrumentos aeronáuticos ou inexperiência.

No tipo "FATOR MATERIAL" estão os aspectos de falhas mecânicas, enquanto no tipo "OUTRO" são encontrados fatores como influência do meio-ambiente.


Finalmente, é possível observar que o fator que mais causou acidentes dentro do banco de dados verificado foi "JULGAMENTO DE PILOTAGEM".

In [31]:
fig = px.bar(count_tipo_fator.head(10), x='Tipo', y='Ocorrências', labels={'index': 'Tipo', 'fator_area': 'Número de Ocorrências'},
             title='Número de Ocorrências por tipo', width=800, height=750)
fig.show()

In [32]:
fig = px.bar(count_fator.head(10), x='Fator', y='Ocorrências', labels={'index': 'Fator', 'fator_nome': 'Número de Ocorrências'},
             title='Número de Ocorrências por Fator Específico', width=1000, height=700)
fig.update_xaxes(tickangle=45)

fig.show()

#Análise por Quantidade de Aeronáves Envolvidas

In [33]:
df_aeronaves_envolvidas = pd.DataFrame(tb1, columns=['codigo_ocorrencia', 'total_aeronaves_envolvidas'])
df_aeronaves_envolvidas = df_aeronaves_envolvidas.drop_duplicates()
count_aeronaves_envolvidas = df_aeronaves_envolvidas['total_aeronaves_envolvidas'].value_counts().reset_index()
count_aeronaves_envolvidas = count_aeronaves_envolvidas.rename(columns={'index': 'aeronaves_envolvidas', 'total_aeronaves_envolvidas': 'ocorrencias'})

print("Frequências absolutas do total de aeronáves envolvidas na ocorrência:")
display(count_aeronaves_envolvidas)

Frequências absolutas do total de aeronáves envolvidas na ocorrência:


aeronaves_envolvidas  ocorrencias
0                     1         1239
1                     2           13

In [34]:
aeronaves_envolvidas_mais_de_1 = tb1.loc[tb1['total_aeronaves_envolvidas'] > 1]
print(aeronaves_envolvidas_mais_de_1['codigo_ocorrencia'].unique())

[44944 45222 46255 50801 51653 53104 65921 77781 77803 78031 78411 78799
 79109]


É Perceptível, com base na análise dos dados, que a maior parte das ocorrências se deram com o envolvimento de uma aeronave apenas. No caso, as únicas ocorrências em que houve a participação de mais de uma aeronave -- aliás, de duas apenas -- foram as ocorrencias de códigos: 44944, 45222, 46255, 50801, 51653, 53104, 65921, 77781, 77803, 78031, 78411, 78799 e 79109.

In [35]:
fig = px.bar(count_aeronaves_envolvidas.head(10), x='aeronaves_envolvidas', y='ocorrencias', labels={'aeronaves_envolvidas': 'Aeronáves Envolvidas', 'ocorrencias': 'Número de Ocorrências'},
             title='Frequências absolutas do total de aeronáves envolvidas na ocorrência', width=1000, height=700)
fig.update_xaxes(tickangle=45)
fig.show()

# Análise por fase de operação

In [36]:
df_fase = pd.DataFrame(tb1, columns=['aeronave_fase_operacao'])

count_fase = df_fase['aeronave_fase_operacao'].value_counts().reset_index()
count_fase = count_fase.rename(columns={'index': 'Fase', 'aeronave_fase_operacao': 'Ocorrências'})

print("Número de ocorrências por fase:\n")
print(count_fase.head(20).to_string(index=False))




Número de ocorrências por fase:

               Fase  Ocorrências
          DECOLAGEM         1036
              POUSO          917
           CRUZEIRO          681
 CORRIDA APÓS POUSO          549
            MANOBRA          374
      ESPECIALIZADA          276
  APROXIMAÇÃO FINAL          242
             SUBIDA          187
            DESCIDA          154
CIRCUITO DE TRÁFEGO          120
               TÁXI           95
      INDETERMINADA           90
         OUTRA FASE           86
 VOO A BAIXA ALTURA           76
 ARREMETIDA NO SOLO           62
   ARREMETIDA NO AR           53
            PAIRADO           46
 DECOLAGEM VERTICAL           36
   OPERAÇÃO DE SOLO           29
     ESTACIONAMENTO           27


A análise realizada nessa parte do banco de dados deixa claro que a fase da operação em que mais ocorreram acidentes foi a fase de "DECOLAGEM".

In [37]:
fig = px.bar(count_fase.head(20), x='Fase', y='Ocorrências', labels={'index': 'Fase', 'aeronave_fase_operacao': 'Número de Ocorrências'},
             title='Número de Ocorrências por Fase', width=1000, height=700)
fig.update_xaxes(tickangle=45)

fig.show()

#Análise pelo Dano da Aeronave

In [38]:
df_nivel_dano = pd.DataFrame(tb1, columns=['codigo_ocorrencia', 'aeronave_nivel_dano'])
df_nivel_dano = df_nivel_dano.drop_duplicates()
count_nivel_dano = df_nivel_dano['aeronave_nivel_dano'].value_counts().reset_index()
count_nivel_dano = count_nivel_dano.rename(columns={'index': 'nivel_dano', 'aeronave_nivel_dano': 'ocorrencias'})

print("Quantidade de ocorrências para cada nível de dano:")
display(count_nivel_dano)

Quantidade de ocorrências para cada nível de dano:


nivel_dano  ocorrencias
0  SUBSTANCIAL          847
1    DESTRUÍDA          172
2         LEVE          140
3       NENHUM           80
4          ***           14

Percebe-se, pela análise dos dados, que, na maioria das ocorrências, o dano na aeronave foi substancial; em segundo lugar, quanto à frequência, há casos em que a aeronave foi destruída; em terceiro lugar, casos em que houve dano leve; em quarto lugar, caso em que não houve dano algum; por fim, casos em que essa informação é ausente -- possivelmente porque a aeronave encontra-se perdida.

In [39]:
fig = px.bar(count_nivel_dano.head(10), x='nivel_dano', y='ocorrencias', labels={'nivel_dano': 'Nível de Dano', 'ocorrencias': 'Número de Ocorrências'},
             title='Quantidade de ocorrências para cada nível de dano', width=1000, height=700)
fig.update_xaxes(tickangle=45)
fig.show()